toolchain/check/impl_validation.cpp

```diff
@@ -22,15 +22,9 @@ namespace Carbon::Check {
 
 namespace {
 
-struct ImplAndInterface {
-  SemIR::ImplId impl_id;
-  SemIR::SpecificInterface interface;
-};
-
-// We avoid holding a reference into the ImplStore, as the act of verifying the
-// diagnostic checks can invalidate such a reference. We collect what we need to
-// know about the the `Impl` for validation into this struct.
+// All information about a `SemIR::Impl` needed for validation.
```

Comment:
Noting the general take here, why doesn't this keep a reference to `impl` in place of is_final, witness_id, self_id, etc?

The prior comment was indicating this was primarily to avoid a reference, and `GetImplInfo` is mainly copying values.

Reply:
Yeah, the structure was originally written to avoid holding a reference but now it holds both info from the Impl as well as other information about it, like the ImplId and TypeStructure, so it has a larger use in how the code is structured now.

Reply:
Sure, there's structured information, but what about replacing:

```
          .is_final = impl.is_final,
          .witness_id = impl.witness_id,
          .self_id = impl.self_id,
          .latest_decl_id = impl.latest_decl_id(),
          .interface = impl.interface,
```

With just `.impl = impl,`